xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -78,6 +80,7 @@
   private XdsClient.LdsResourceWatcher listenerWatcher;
   private boolean newServerApi;
   @VisibleForTesting final Set<ServerWatcher> serverWatchers = new HashSet<>();
+  private final Object lock = new Object();
```

Comment:
Have you considered using `AtomicReference` instead of using this `lock` and `synchronized` blocks?

Reply:
yea, it should look less cumbersome.